File: P1/Sesion1/FactoriaAbstracta_Prototipos/src/prototipo/BicicletaCarretera.java
package prototipo;

public class BicicletaCarretera extends Bicicleta {

    private static int COUNT = 0;

    /**
     * 
     */
    public BicicletaCarretera() {
        BicicletaCarretera.COUNT += 1;
        tipo = Tipo.CARRETERA;
        id = COUNT;
    }

}

File: P1/Sesion1/FactoriaAbstracta_Prototipos_RUBY/lib/bicicleta_carretera.rb
module FactoriaAbstracta_Prototipos_RUBY
  
  class BicicletaCarretera < Bicicleta

    #Variable de clase
    @@count = 0

    attr_accessor :prototype

    def initialize(prototype)
      @prototype = prototype
      @@count += 1 
      @tipo = Tipo::CARRETERA
      @id = @@count
    end

  end
  
end
